test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -282,6 +289,88 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 					framework.ExpectError(err)
 				}
 			})
+
+			ginkgo.It("while original volume is still in use, create snapshot, create new pvc from snapshot, create new pod to use the pvc on the same node, validate data on volume", func() {
+				testSnapshotWithOriginalVolumeInUse(pattern, f, cs, dc, pvc, config, sc, claimSize,
+					createAndTestPodClaimPV, originalMntTestData, nodeName, true, /* sameNode */
+					config.ClientNodeSelection, cleanupSteps)
+			})
+
+			ginkgo.It("while original volume is still in use, create snapshot, create new pvc from snapshot, create new pod to use the pvc on a different node, validate data on volume", func() {
+				testSnapshotWithOriginalVolumeInUse(pattern, f, cs, dc, pvc, config, sc, claimSize,
+					createAndTestPodClaimPV, originalMntTestData, nodeName, false, /* sameNode */
+					config.ClientNodeSelection, cleanupSteps)
+			})
 		})
 	})
 }
+
+func createSnapshot(f *framework.Framework, config *storageframework.PerTestConfig, pattern storageframework.TestPattern, pvc *v1.PersistentVolumeClaim, cleanupSteps []func()) (vs, vscontent, vsc *unstructured.Unstructured) {
+	var sr *storageframework.SnapshotResource
+	cleanupSteps = append(cleanupSteps, func() {
+		framework.ExpectNoError(sr.CleanupResource(f.Timeouts))
+	})
+	sr = storageframework.CreateSnapshotResource(sDriver, config, pattern, pvc.GetName(), pvc.GetNamespace(), f.Timeouts)
+	return sr.Vs, sr.Vscontent, sr.Vsclass
+}
+
+func testSnapshotWithOriginalVolumeInUse(pattern storageframework.TestPattern, f *framework.Framework, cs clientset.Interface, dc dynamic.Interface, pvc *v1.PersistentVolumeClaim,
+	config *storageframework.PerTestConfig, sc *storagev1.StorageClass, claimSize string, createAndTestPodClaimPV func(bool),
+	originalMntTestData string, nodeName string, requireSameNode bool, node e2epod.NodeSelection, cleanupSteps []func()) {
```

Comment:
@msau42 What are your thoughts on keeping this function inside the `DefineTests` function similar to how the `init` and `createSnapshot` functions are? That would remove the issue of passing 14-15 parameters like I am having to do it now.

Reply:
@dr0pdb the currently function does have too many parameters. Could you simplify it if possible? Thanks!